src/m365/commands/request.ts

```diff
@@ -0,0 +1,187 @@
+      };
+
+      const unknownOptions: any = this.getUnknownOptions(args.options);
+      const unknownOptionsNames: string[] = Object.getOwnPropertyNames(unknownOptions);
+      unknownOptionsNames.forEach(o => {
+        properties[o] = typeof unknownOptions[o] !== 'undefined';
+      });
+
+      Object.assign(this.telemetryProperties, properties);
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --url <url>'
+      },
+      {
+        option: '-m, --method [method]',
+        autocomplete: this.allowedMethods
+      },
+      {
+        option: '-r, --resource [resource]'
+      },
+      {
+        option: '-b, --body [body]'      
+      },
+      {
+        option: '-p, --filePath [filePath]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const currentMethod = args.options.method || 'get';
+        if (this.allowedMethods.indexOf(currentMethod) === -1) {
+          return `${currentMethod} is not a valid value for method. Allowed values: ${this.allowedMethods.join(', ')}`;
+        }
+
+        if (args.options.body && (!args.options.method || args.options.method === 'get')) {
+          return 'Specify a different method when using body';
+        }
+
+        if (args.options.body && !args.options['content-type']) {
+          return 'Specify the content-type when using body';
+        }
+
+        if (args.options.filePath && !fs.existsSync(path.dirname(args.options.filePath))) {
+          return 'The location specified in the filePath does not exist';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.debug) {
+      logger.logToStderr(`Preparing request...`);
+    }
+
+    const method = (args.options.method || 'get').toUpperCase();
+    const headers: AxiosRequestHeaders = {};
+
+    const unknownOptions: any = this.getUnknownOptions(args.options);
+    const unknownOptionsNames: string[] = Object.getOwnPropertyNames(unknownOptions);
+    unknownOptionsNames.forEach(o => {
+      headers[o] = unknownOptions[o];
+    });
+
+    if (!headers.accept) {
+      headers.accept = 'application/json';
+    }
+
+    if (args.options.resource) {
+      headers['x-resource'] = args.options.resource;
+    }
+
+    const config: AxiosRequestConfig<string> = {
+      url: args.options.url,
+      headers,
+      method,
+      data: args.options.body
+    };
+
+    if (headers.accept.toString().startsWith('application/json')) {
+      config.responseType = 'json';
+    }
+
+    if (args.options.filePath) {
+      config.responseType = 'stream';
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Executing request...`);
+    }
+
+    if (args.options.filePath) {
+      request
+        .execute<any>(config)
+        .then((file: any): Promise<string> => {
```

Comment:
Instead of using `any`, let's use `AxiosResponse` so that we have typing for the `data` property